cmd/noisd/genesis.go

```diff
@@ -0,0 +1,381 @@
+	}
+	authGenState.Accounts = genAccs
+	authGenStateBz, err := cdc.MarshalJSON(&authGenState)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to marshal auth genesis state: %w", err)
+	}
+	appState[authtypes.ModuleName] = authGenStateBz
+
+	// save balances
+	bankGenState.Balances = banktypes.SanitizeGenesisBalances(balances)
+	bankGenStateBz, err := cdc.MarshalJSON(bankGenState)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to marshal bank genesis state: %w", err)
+	}
+	appState[banktypes.ModuleName] = bankGenStateBz
+
+	// alloc module genesis
+	allocationGenState := allocationtypes.GetGenesisStateFromAppState(cdc, appState)
+	allocationGenState.Params = genesisParams.AllocationParams
+	allocationGenStateBz, err := cdc.MarshalJSON(allocationGenState)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to marshal alloc genesis state: %w", err)
+	}
+	appState[allocationtypes.ModuleName] = allocationGenStateBz
+
+	// wasm
+	// wasm module genesis
+	wasmGenState := &wasm.GenesisState{
+		Params: genesisParams.WasmParams,
+	}
+	wasmGenStateBz, err := cdc.MarshalJSON(wasmGenState)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to marshal wasm genesis state: %w", err)
+	}
+	appState[wasm.ModuleName] = wasmGenStateBz
+
+	return appState, genDoc, nil
+}
+
+// params only
+func MainnetGenesisParams() GenesisParams {
+	genParams := GenesisParams{}
+
+	genParams.GenesisTime = time.Now()
+
+	genParams.NativeCoinMetadatas = []banktypes.Metadata{
+		{
+			Description: "The native token of Nois Network",
+			DenomUnits: []*banktypes.DenomUnit{
+				{
+					Denom:    BaseCoinUnit,
+					Exponent: 0,
+					Aliases:  nil,
+				},
+				{
+					Denom:    HumanCoinUnit,
+					Exponent: NoisExponent,
+					Aliases:  nil,
+				},
+			},
+			Name:    "NOIS",
+			Base:    BaseCoinUnit,
+			Display: HumanCoinUnit,
+			Symbol:  "NOIS",
+		},
+	}
+
+	// alloc
+	genParams.AllocationParams = allocationtypes.DefaultParams()
+	genParams.AllocationParams.DistributionProportions = allocationtypes.DistributionProportions{
+		RandomnessRewards: sdk.NewDecWithPrec(4, 2),  // 4%
+		ValidatorRewards:  sdk.NewDecWithPrec(6, 2),  // 6%
+		DeveloperRewards:  sdk.NewDecWithPrec(20, 2), // 20 %
+	}
+
+	genParams.AllocationParams.WeightedDeveloperRewardsReceivers = []allocationtypes.WeightedAddress{}
+	// example
+	// genParams.AllocationParams.WeightedDeveloperRewardsReceivers = []allocationtypes.WeightedAddress{
+	//	{
+	//		Address: "nois1....",
+	//		Weight:  sdk.OneDec(), // 100%
+	//  },
+	// }
+	genParams.AllocationParams.RandomnessRewardsReceiver = ""
+
+	// mint
+	genParams.MintParams = minttypes.DefaultParams()
+	genParams.MintParams.MintDenom = BaseCoinUnit
+	//  default mint params change accordingly
+	genParams.MintParams.InflationRateChange = sdk.NewDecWithPrec(13, 2)
+	genParams.MintParams.InflationMax = sdk.NewDecWithPrec(20, 2)
+	genParams.MintParams.InflationMin = sdk.NewDecWithPrec(7, 2)
```

Comment:
```suggestion
	genParams.MintParams.InflationMax = sdk.NewDecWithPrec(20, 2)
	genParams.MintParams.InflationMin = sdk.NewDecWithPrec(20, 2)
```

Let's make this 20% fixed. I checked and got confirmation min and max are allowed to have the same value